src/config/config.c

```diff
@@ -378,3 +476,54 @@ int ReadConfig(int modules, const char *cfgfile, void *d1, void *d2)
     OS_ClearXML(&xml);
     return (0);
 }
+
+int Test_Analysisd(const char *path, char **output) {
+    int fail = 0;
+    _Config *test_config;
+    os_calloc(1, sizeof(_Config), test_config);
+
+    int modules = CGLOBAL | CRULES | CALERTS | CCLUSTER;
+    if( (ReadConfig(modules, path, test_config, NULL, output) < 0) || (ReadConfig(CLABELS, path, &(test_config->labels), NULL, output) < 0) ) {
+        if (output == NULL){
+            merror(CONF_READ_ERROR, "Analysisd");
+        } else {
+            wm_strcat(output, "ERROR: Invalid configuration in Analysisd", '\n');
+        }
+        fail = 1;
+    }
+
+    if(!fail) {
+        test_config->min_rotate_interval = getDefine_Int("analysisd", "min_rotate_interval", 10, 86400);
+
+        if (test_config->rotate_interval && (test_config->rotate_interval < test_config->min_rotate_interval || test_config->rotate_interval > 86400)) {
+            if (output == NULL){
+                merror("Rotate interval setting must be between %d seconds and one day.", test_config->min_rotate_interval);
+            } else {
+                wm_strcat(output, "ERROR: Invalid configuration in Analysisd", '\n');
+            }
+            fail = 1;
+        }
+
+        if (!fail && test_config->max_output_size && (test_config->max_output_size < 1000000 || test_config->max_output_size > 1099511627776)) {
```

Comment:
The condition `!fail` has already been evaluated previously. Please, removed it.